### PR TITLE
Add ChangeStartupMsg test for platform test

### LINKED ISSUE #78: Platform test setting the port of all services to the same value
https://github.com/canonical/edgex-snap-testing/blob/d61a56465cb65a39e2f711f805e865b27643ed82/test/suites/edgexfoundry/main_test.go#L72-L73

These should be set to false or removed in favor of another custom test that doesn't mess with a port globally.

### PR DESCRIPTION
**Pull request by MonicaisHer**
resolves #78

### CODE REVIEW

**Review by farshidtz — CHANGES_REQUESTED** [review]
It looks good overall. I've commented regarding some minor issues.

As a discussion point, I am thinking if this test should be default for all suites, or turned into a common test and ran selectively. It is relevant for other multi-service snaps too, currently also edgex-device-usb, even though it doesn't cause a conflict like for edgexfoundry.

**Comment by farshidtz** [comment]
Actually the test will pass as long as the script in command chain exports any variable with that value. For example, you can set X=<message> and it will still pass.

I think we should be looking for the `msg=<message>`.

```
2022-07-26T13:11:27+02:00 edgex-device-mqtt.device-mqtt[44970]: ++ export SERVICE_STARTUPMSG=test
...
2022-07-26T13:11:27+02:00 edgex-device-mqtt.device-mqtt[44970]: level=INFO ts=2022-07-26T11:11:27.153627777Z app=device-mqtt source=variables.go:352 msg="Variables override of 'Service.StartupMsg' by environment variable: SERVICE_STARTUPMSG=test"
...
2022-07-26T13:12:27+02:00 edgex-device-mqtt.device-mqtt[44970]: level=INFO ts=2022-07-26T11:12:27.199253366Z app=device-mqtt source=message.go:55 msg=test
```

File: test/suites/edgexfoundry/config_test.go [hunk 1]
```diff
@@ -34,3 +39,96 @@ func TestEnvConfigChangePort(t *testing.T) {
...
+	// start clean
+	utils.SnapStop(t, supportSchedulerService)
+
+	t.Cleanup(func() {
+		utils.SnapUnset(t, platformSnap, startupMsgKey)
+		utils.SnapStop(t, supportSchedulerService)
+	})
+
+	// enable new config option to avoid mixed options issue with old env option
+	utils.SnapSet(t, platformSnap, "app-options", "true")
```
Comment (farshidtz) [inline test/suites/edgexfoundry/config_test.go:60]: We turn it on to enable app options, which are being tested.

File: test/suites/edgexfoundry/config_test.go [hunk 2]
```diff
@@ -34,3 +39,96 @@ func TestEnvConfigChangePort(t *testing.T) {
...
+
+	// make sure the startupmsg is the default testing message before setting it
+	utils.SnapStart(t, supportSchedulerService)
+	require.True(t, checkStartupMsg(t, supportSchedulerService, defaultStartupMsg))
```
Comment (farshidtz) [inline test/suites/edgexfoundry/config_test.go:64]: The error message is hard to follow when this fails. Better add a message instead of comment in code.

**Comment by farshidtz** [comment]
@MonicaisHer okay with the changes?

**Review by farshidtz — APPROVED** [review]

**Comment by MonicaisHer** [comment]
> @MonicaisHer okay with the changes?

thanks for your changes, the changes look great!
